api/controllers/collection.js

```diff
@@ -0,0 +1,392 @@
+"use strict";
+const express = require("express");
+const cache = require("apicache");
+const fs = require("fs");
+
+const {
+  db,
+  as,
+  CREATE_COLLECTION,
+  COLLECTION_BY_ID,
+  INSERT_AUTHOR,
+  INSERT_LOCALIZED_TEXT,
+  UPDATE_COLLECTION,
+  UPDATE_AUTHOR_FIRST,
+  UPDATE_AUTHOR_LAST,
+  FEATURED_MAP,
+  ENTRIES_BY_COLLECTION_ID,
+  refreshSearch,
+  ErrorReporter,
+} = require("../helpers/db");
+
+const {
+  setConditional,
+  maybeUpdateUserText,
+  parseGetParams,
+  validateUrl,
+  verifyOrUpdateUrl,
+  returnByType,
+  fixUpURLs
+} = require("../helpers/things");
+
+const logError = require("../helpers/log-error.js");
+
+const requireAuthenticatedUser = require("../middleware/requireAuthenticatedUser.js");
+const COLLECTION_STRUCTURE = JSON.parse(
+  fs.readFileSync("api/helpers/data/collection-structure.json", "utf8")
+);
+
+// strip off final character (assumed to be "s")
+const singularLowerCase = name =>
+  (name.slice(-1) === "s" ? name.slice(0, -1) : name).toLowerCase();
+
+// just get the type, if specified
+const typeFromReq = req => {
+  var cat = singularLowerCase(req.query.selectedCategory || "Alls");
+  let selectedCategoryValues = ['all', 'case', 'method', 'organization'];
+  if (selectedCategoryValues.indexOf(cat) < 0) {
+    cat = 'all';
+  }
+  return cat === "all" ? "thing" : cat;
+};
+
+function randomTexture() {
+  let index = Math.floor(Math.random() * 6) + 1;
+  return `/images/texture_${index}.svg`;
+}
+
+/**
```

Comment:
all of these api comments are not up-to-date/accurate. i feel like we should remove them all to keep things cleaner.